src/main/java/baritone/pathing/movement/movements/MovementDescend.java

```diff
@@ -101,7 +101,7 @@ public static void cost(CalculationContext context, int x, int y, int z, int des
         //C, D, etc determine the length of the fall
 
         IBlockState below = context.get(destX, y - 2, destZ);
-        if (!MovementHelper.canWalkOn(context.bsi, destX, y - 2, destZ, below)) {
+        if (!MovementHelper.canWalkOn(context.bsi, destX, y - 2, destZ, below) || (below.getBlock() == Blocks.MAGMA && !context.allowSneakOverMagma)) {
```

Comment:
Doesn't `canWalkOn` already return false for magma if `allowSneakOverMagma` is false?